### PR TITLE
Add generalized electronic response function

### PR DESCRIPTION
**Pull request by cnhunt**

### CODE REVIEW

File: AtDigitization/AtDigiLinkDef.h [hunk 1]
```diff
@@ -13,5 +13,6 @@
 #pragma link C++ class AtSimulatedLine + ;
 #pragma link C++ class AtTrigger + ;
 #pragma link C++ class AtTriggerTask + ;
+#pragma link C++ class AtVectorResponse + ;
```
Comment (anthoak13) [inline AtDigitization/AtDigiLinkDef.h:16]: This never has to be written to disk, so should be `#pragma link C++ class AtVectorResponse - !;`

File: AtDigitization/AtVectorResponse.h [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+#include <FairRuntimeDb.h>
+#include <FairTask.h>
+
+#include <Rtypes.h>
+#include <TString.h>
+
+#include <cmath>
+#include <cstdio>
+#include <iostream>
+#include <vector>
```
Comment (anthoak13) [inline AtDigitization/AtVectorResponse.h:16]: I think this should be the only header file here (along with Rtypes.h). Everything else is not used in the class definition

File: AtDigitization/AtVectorResponse.h [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+#define ATVECTORRESPONSE_H
+
+#include "AtDigiPar.h"
+
+#include <FairRunAna.h>
+#include <FairRuntimeDb.h>
+#include <FairTask.h>
+
+#include <Rtypes.h>
+#include <TString.h>
```
Comment (anthoak13) [inline AtDigitization/AtVectorResponse.h:11]: Should be a forward declaration, not an include

File: AtDigitization/AtVectorResponse.cxx [hunk 4]
```diff
@@ -0,0 +1,43 @@
+#include "AtVectorResponse.h"
```
Comment (anthoak13) [inline AtDigitization/AtVectorResponse.cxx:1]: Includes not needed in the definition of the class should be moved here

**Review by anthoak13 — CHANGES_REQUESTED** [review]
Two changes of note:

This isn't a ROOT class, so we just want to generate the parts of the dictionary for use in the terminal (thing rootmap) so the LinkDef needs tweaking.

Most of the includes in the header file should be in the source file. (You can run iwyu to verify).

File: AtDigitization/AtVectorResponse.h (hunk 2 above)
Comment (anthoak13) [reply AtDigitization/AtVectorResponse.h:16]: This is pointing to the vector include.

File: AtDigitization/AtVectorResponse.h (hunk 3 above)
Comment (anthoak13) [reply AtDigitization/AtVectorResponse.h:11]: Pointing to TString

**Comment by anthoak13** [comment]
@cnhunt The changes can be made as a separate commit. I can squash when merging in.

File: AtDigitization/AtDigiLinkDef.h [hunk 5]
```diff
@@ -13,5 +13,6 @@
 #pragma link C++ class AtSimulatedLine + ;
 #pragma link C++ class AtTrigger + ;
 #pragma link C++ class AtTriggerTask + ;
+#pragma link C++ class AtVectorResponse - ;
```
Comment (anthoak13) [inline AtDigitization/AtDigiLinkDef.h:16]: This should probably have a "!" also.

File: AtDigitization/AtVectorResponse.cxx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+#include "AtDigiPar.h"
+
+#include <FairRunAna.h>
+#include <FairRuntimeDb.h>
+#include <FairTask.h>
```
Comment (anthoak13) [inline AtDigitization/AtVectorResponse.cxx:7]: Remove

File: AtDigitization/AtVectorResponse.cxx [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
```
Comment (anthoak13) [inline AtDigitization/AtVectorResponse.cxx:8]: Should add the following:
image

File: AtDigitization/AtVectorResponse.h [hunk 8]
```diff
@@ -0,0 +1,26 @@
+#ifndef ATVECTORRESPONSE_H
+#define ATVECTORRESPONSE_H
+
+#include <Rtypes.h>
+#include <TString.h>
+
+#include <cmath>
```
Comment (anthoak13) [inline AtDigitization/AtVectorResponse.h:7]: Remove

File: AtDigitization/AtVectorResponse.h [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+#include <cstdio>
```
Comment (anthoak13) [inline AtDigitization/AtVectorResponse.h:8]: Remove

File: AtDigitization/AtVectorResponse.h [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+#include <iostream>
```
Comment (anthoak13) [inline AtDigitization/AtVectorResponse.h:9]: Remove

File: AtDigitization/AtVectorResponse.cxx [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+Double_t AtVectorResponse::fPeakingTime = 0;
+Double_t AtVectorResponse::fTBTime = 0;
+std::vector<Double_t> AtVectorResponse::fWaveSample = {};
+
+AtVectorResponse::AtVectorResponse() {}
```
Comment (anthoak13) [inline AtDigitization/AtVectorResponse.cxx:13]: Prefer `= default` over trivial constructor (or just don't define)

**Review by anthoak13 — CHANGES_REQUESTED** [review]

File: AtDigitization/AtDigiLinkDef.h (hunk 5 above)
Comment (cnhunt) [reply AtDigitization/AtDigiLinkDef.h:16]: where should the "!" be? What is it?

File: AtDigitization/AtVectorResponse.cxx (hunk 6 above)
Comment (cnhunt) [reply AtDigitization/AtVectorResponse.cxx:7]: IDE disagrees with that action.

File: AtDigitization/AtDigiLinkDef.h (hunk 5 above)
Comment (anthoak13) [reply AtDigitization/AtDigiLinkDef.h:16]: `#pragma link C++ class AtVectorResponse - !;`

It disables the generation of a `<<` operator.

File: AtDigitization/AtVectorResponse.cxx (hunk 6 above)
Comment (anthoak13) [reply AtDigitization/AtVectorResponse.cxx:7]: I disagree with the IDE, it compiles with FairTask removed.

File: AtDigitization/AtVectorResponse.cxx [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+   if (!input.is_open())
+      std::cout << "wave input not open" << std::endl;
+   if (input.is_open()) {
+      std::cout << "wave input is open" << std::endl;
+      while (!input.eof()) {
+         double value = 0;
+         input >> value;
+         fWaveSample.push_back(value);
+      }
+   }
```
Comment (anthoak13) [inline AtDigitization/AtVectorResponse.cxx:34]: Test thing

**Review by anthoak13 — APPROVED** [review]

**Pull request merged**